web/packages/teleterm/src/ui/HeadlessAuthn/HeadlessPrompt/HeadlessPrompt.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import * as Alerts from 'design/Alert';
+import { ButtonIcon, Text, ButtonSecondary } from 'design';
+import DialogConfirmation, {
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+} from 'design/DialogConfirmation';
+import { Attempt } from 'shared/hooks/useAsync';
+import * as Icons from 'design/Icon';
+
+import { PromptWebauthn } from '../../ClusterConnect/ClusterLogin/FormLogin/PromptWebauthn';
+
+import type * as tsh from 'teleterm/services/tshd/types';
+
+export type HeadlessPromptProps = {
+  cluster: tsh.Cluster;
+  clientIp: string;
+  onApprove(): Promise<void>;
+  onReject(): Promise<void>;
+  updateHeadlessStateAttempt: Attempt<void>;
+  onCancel(): void;
+};
+
+export function HeadlessPrompt({
+  cluster,
+  clientIp,
+  onApprove,
+  onReject,
+  updateHeadlessStateAttempt,
+  onCancel,
+}: HeadlessPromptProps) {
+  const [waitForMfa, setWaitForMfa] = useState(false);
+
+  return (
+    <DialogConfirmation
+      dialogCss={() => ({
+        maxWidth: '480px',
+        width: '100%',
+      })}
+      disableEscapeKeyDown={false}
+      open={true}
+    >
+      <DialogHeader justifyContent="space-between" mb={0} alignItems="baseline">
+        <Text typography="h4">
+          Headless login to <b>{cluster.name}</b>
```

Comment:
Should we call it headless login if it's not technically a login? I see that the Web UI avoids this terminology altogether. Perhaps it should say "Headless request"?

Reply:
I think `headless command` is better. We don't want to conflate headless with access requests. Does "Headless command on {cluster}" sound good?

Reply:
Yeah, it sounds good!